RNTester/js/CameraRollView.js

```diff
@@ -238,10 +201,15 @@ class CameraRollView extends React.Component<Props, State> {
 
     if (assets.length > 0) {
       newState.lastCursor = data.page_info.end_cursor;
-      newState.assets = this.state.assets.concat(assets);
-      newState.dataSource = this.state.dataSource.cloneWithRows(
-        groupByEveryN(newState.assets, this.props.imagesPerRow),
-      );
+
+      // Dedup assets we want to append
+      newState.assets = this.state.assets
+        .concat(assets)
+        .filter(
+          (a1, index, self) =>
+            index ===
+            self.findIndex(a2 => a1.node.image.uri === a2.node.image.uri),
```

Comment:
This algorithm is O(n*n). Can you instead do `Array.from(new Set(this.state.assets.concat(assets))`?